File: app/src/main/java/de/werkelmann/rekote/view/dialogs/ErrorDialog.java
package de.werkelmann.rekote.view.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import de.werkelmann.rekote.R;

public class ErrorDialog extends DialogFragment {

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return build();
    }

    public Dialog build() {
        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        builder.setMessage(getString(R.string.trivial_error_message));
        return builder.create();
    }

}

File: app/src/main/kotlin/de/werkelmann/rekote/view/dialogs/ErrorDialog.kt
package de.werkelmann.rekote.view.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import de.werkelmann.rekote.R

class ErrorDialog : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return build()
    }

    fun build(): Dialog {
        val builder = AlertDialog.Builder(context)
        builder.setMessage(getString(R.string.trivial_error_message))
        return builder.create()
    }

}
